client/x/evmstaking/keeper/withdrawal_queue.go

```diff
@@ -17,12 +17,10 @@ func (k Keeper) AddWithdrawalToQueue(ctx context.Context, withdrawal types.Withd
 	return k.WithdrawalQueue.Enqueue(ctx, withdrawal)
 }
 
-func (k Keeper) DequeueEligibleWithdrawals(ctx context.Context) (withdrawals etypes.Withdrawals, err error) {
-	maxDequeue, err := k.MaxWithdrawalPerBlock(ctx)
-	if err != nil {
-		return nil, err
+func (k Keeper) DequeueEligibleWithdrawals(ctx context.Context, maxDequeue int) (withdrawals etypes.Withdrawals, err error) {
```

Comment:
Why don't we add break condition when looping the queue? If the queue is empty, just break the loop.